src/services/formatting/smartIndenter.ts

```diff
@@ -436,7 +436,6 @@ namespace ts.formatting {
                 case SyntaxKind.Parameter:
                 case SyntaxKind.FunctionType:
                 case SyntaxKind.ConstructorType:
-                case SyntaxKind.UnionType:

```

Comment:
I removed it to prevent this:
```typescript
type a = 
    string |
        number |
        boolean;
        ;
```
... to match normal bar operator behavior:
```typescript
let x = 
    3 | 
    4 |
    5;
```